picard/tagger.py

```diff
@@ -321,8 +322,36 @@ def pipe_server(self):
         while self.pipe_handler.pipe_running:
             messages = [x for x in self.pipe_handler.read_from_pipe() if x not in IGNORED]
             if messages:
-                self.add_paths(messages)
-                self.bring_tagger_front()
+                self.load_to_picard(messages)
+
+    def load_to_picard(self, items):
+        files = set()
+        mbids = set()
+        urls = set()
+
+        for item in items:
+            parsed = urlparse(item)
+            if not parsed.scheme:
+                files.add(item)
+            elif parsed.scheme == "file":
+                # remove file:// prefix safely
+                files.add(item[7:])
+            elif parsed.scheme in {"http", "https"}:
+                # .path returns / before actual link
+                urls.add(parsed.path[1:])
+            elif parsed.scheme == "mbid":
+                mbids.add(parsed.netloc + parsed.path)
```

Comment:
To ease testing and to keep functions short, it would be preferable if this part of the code was in its own method.

```python
@staticmethod
def _parse_file_or_url_items(items):
    files = set()
    mbids = set()
    urls = set()

    ....

    return files, mbids, urls

def load_to_picard(self, items):
    files, mbids, urls = self._parse_file_or_url_items(items)
```

Then we can add tests for this method alone.

Reply:
Seems nice, thanks. Though I'd rather return a dict instead of `return files, mbids, urls`. We're planning to parse commands as well in the future, four returned values seem to be more error-prone and overcomplicated.